src/integration-test/java/com/mongodb/jdbc/integration/testharness/IntegrationTestUtils.java

```diff
@@ -553,11 +578,15 @@ private static boolean compareRow(List<Object> expectedRow, ResultSet actualRow)
                     /*
                     case Types.OTHER:
                         if (expectedRow.get(i) != actualRow.getObject(i + 1)) {
+                            System.err.println("Expected " + expected + " but is " + actual);
                             return false;
                         }
                      */
                 case Types.OTHER:
+                    Object expected_obj = expectedRow.get(i);
+                    Object actual_obj = expectedRow.get(i);
                     if (!expectedRow.get(i).equals(actualRow.getObject(i + 1))) {
```

Comment:
To follow your previous edits, and change to get the actual_obj:
```suggestion
                    Object actual_obj = actualRow.getObject(i + 1);
                    if (!expected_obj.equals(actual_obj)) {
```

Reply:
oh, thank you for spotting this. Definitely a mistake, I forgot to update after asking you if this was supposed to be this way or not.